gammapy/data/observations.py

```diff
@@ -346,51 +346,65 @@ def check(self, checks="all"):
         checker = ObservationChecker(self)
         return checker.run(checks=checks)
 
-    def peek(self, figsize=(12, 10)):
+    def peek(self, figsize=(15, 10)):
         """Quick-look plots in a few panels.
 
         Parameters
         ----------
         figsize : tuple
             Figure size
         """
-        n_irfs = len(self.available_hdus)
 
-        fig, axes = plt.subplots(
-            nrows=n_irfs // 2,
-            ncols=2 + n_irfs % 2,
-            figsize=figsize,
-            gridspec_kw={"wspace": 0.25, "hspace": 0.25},
-        )
-
-        axes_dict = dict(zip(self.available_hdus, axes.flatten()))
-
-        if "aeff" in self.available_hdus:
-            self.aeff.plot(ax=axes_dict["aeff"])
-            axes_dict["aeff"].set_title("Effective area")
-
-        if "bkg" in self.available_hdus:
-            bkg = self.bkg
+        plottable_hds = ["events", "aeff", "psf", "edisp", "bkg", "rad_max"]
 
-            if not bkg.has_offset_axis:
-                bkg = bkg.to_2d()
+        plot_hdus = list(set(self.available_hdus) & set(plottable_hds))
 
-            bkg.plot(ax=axes_dict["bkg"])
-            axes_dict["bkg"].set_title("Background rate")
-        else:
-            logging.warning(f"No background model found for obs {self.obs_id}.")
+        n_irfs = len(plot_hdus)
+        nrows = n_irfs // 2
+        ncols = 2 + n_irfs % 2
 
-        if "psf" in self.available_hdus:
-            self.psf.plot_containment_radius_vs_energy(ax=axes_dict["psf"])
-            axes_dict["psf"].set_title("Point spread function")
-        else:
-            logging.warning(f"No PSF found for obs {self.obs_id}.")
+        fig, axes = plt.subplots(
+            nrows=nrows,
+            ncols=ncols,
+            figsize=figsize,
+            gridspec_kw={"wspace": 0.3, "hspace": 0.3},
+        )
 
-        if "edisp" in self.available_hdus:
-            self.edisp.plot_bias(ax=axes_dict["edisp"], add_cbar=True)
-            axes_dict["edisp"].set_title("Energy dispersion")
-        else:
-            logging.warning(f"No energy dispersion found for obs {self.obs_id}.")
+        for idx in range(nrows * ncols):
+            ax = axes.flat[idx]
+            try:
```

Comment:
Please write this `try/except` more compact, because the `IndexError` could happen anywhere in the code block and it would go silent. Ideally `try/except` statements only cover a single line, or maybe just use:
```
for ax, name in zip_longest(axes.flat, plot_hdus):
    if name is None:
        ax.set_visible(False)
        continue
```